### PR TITLE
Add repeatbase_test.

### PR DESCRIPTION
**Pull request by chicocvenancio**
Creates test for accompanying issue. #3609

### CODE REVIEW

**Comment by chicocvenancio** [comment]
I signed it

**Comment by natasha41575** [comment]
Thanks for the test! Please fix the lint errors that the presubmit check found

File: api/krusty/repeatbase_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      - name: container-a
+        image: image-canary-a
+`)
+
+	th.WriteK("/app", `
+resources:
+  - deployment-a
+`)
+
+	m := th.Run("/app", th.MakeDefaultOptions())
```
Comment (natasha41575) [inline api/krusty/repeatbase_test.go:128]: In #3609 you reported that this test will produce an error (and thus will fail here); could you please change this test to 
1) include a comment that it is currently broken
2) use th.RunWithErr instead, such that the output if this test matches what the current output actually is (e.g. the test should "pass" even though it is broken behavior"
3) include a comment of what the expected output should be - or include what the output should be as a comment in the issue you filed?

Comment (chicocvenancio) [reply api/krusty/repeatbase_test.go]: @natasha41575 Thanks for the comments. Hopefully I've addressed all items.

Comment (natasha41575) [reply api/krusty/repeatbase_test.go]: @chicocvenancio awesome, thanks. last couple of things: could you 1) include a link to the issue you filed in the comment documenting that this is broken behavior and 2) squash this PR down to 1 commit?

Comment (chicocvenancio) [reply api/krusty/repeatbase_test.go]: Done!

**Review by natasha41575 — APPROVED** [review]
/lgtm

**Comment by monopole** [comment]
This appear to be
```
  top (emit both canary and base after changing their image names)
   | \
   |  canary (change name of deployment and its image)
   | /
  base (deployment with some image)
```

Bug aside, this sort of diamond pattern is supposed to work (see the tests with `diamond` in the name), but it seems more complex than it needs to be here.  The patches just change image names?  so why not use the image directive?
also, if the base and canary aren't reused anywhere else, the changes at the top aren't needed.
The changes could be pushed down, so all the top level does is combine the base and the canary.

It seems the goal here is to deploy the base and canary at the same time in one apply step?

Suggest something like
```
   top (just combines production and canary)
   /     \
prod  canary (modify base as needed)
  \        /
   base
```

**Comment by chicocvenancio** [comment]
@monopole 
> The patches just change image names? so why not use the image directive?
It also adds a label in the test, and there are additional changes in our use-case. I don't think it matters much if we're using patch or transformers though.

>  if the base and canary aren't reused anywhere else, the changes at the top aren't needed.
In our case they are reused extensively (50+ instances).

**Review by monopole — COMMENTED** [review]
Thanks for sending this.

We trying more to keep these tests more readable as examples, as well as bug reproductions.

Two things:

- Please give objects names that differ from types, e.g. name a `Deployment` something other than `deployment`.  Something random is better, e.g. fruit names.  This makes it easier for new users to understand the examples.

- Use minimal, relative paths to files instead of absolute paths (see below).

If you cannot make time for it let me know, i can merge it and fix it quickly.

**Comment by chicocvenancio** [comment]
@monopole I hope the changes resolve the issues. I can squash once reviewed.

**Comment by monopole** [comment]
/lgtm

thanks!

**Pull request merged**